app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -2141,4 +2143,32 @@ public void onClick(View view) {
     public enum Flag {
         RESET_DATABASE,
     }
+    public void setupNetworkBroadcastReceiver()
+    {
+        BroadcastReceiver networkStateReceiver = new BroadcastReceiver() {
+            final CoordinatorLayout coordinatorLayout = (CoordinatorLayout) findViewById(R.id.coordinatorLayout);
+            @Override
+            public void onReceive(Context context, Intent intent) {
+                Log.d("Network Listener", "Network Type Changed");
+                boolean isConnected = intent.getBooleanExtra(ConnectivityManager.EXTRA_NO_CONNECTIVITY, false);
+                if(isConnected){
+                    //Disable fab when there is no connectivity
+                    fab.setEnabled(false);
```

Comment:
But there can be case where the user is being displayed the chatbox and a network change occurs, 
Maybe just disable the asyncSend button also save the state, like if the user network disconnects and they were writing a message so we can show the chatbox again when the network connects, for a better UX